recipes-bsp/tegra-binaries/tegra-helper-scripts/initrd-flash.sh

```diff
@@ -220,21 +220,40 @@ run_rcm_boot() {
 
 mount_partition() {
     local dev="$1"
-    if udisksctl mount -b "$dev" > /dev/null; then
-	cat /proc/mounts | grep "^$dev" | cut -d' ' -f2
-	return 0
+    local mnt=$(cat /proc/mounts | grep "^$dev" | cut -d' ' -f2)
+    if [ -z ${mnt} ]; then
+        udisksctl mount -b "$dev" > /dev/null
+        local mnt=$(cat /proc/mounts | grep "^$dev" | cut -d' ' -f2)
+    else
+        echo "${mnt}"
+        return 0
+    fi
+    if [ -z ${mnt} ]; then
+        echo "Error: Could not mount ${mnt}!" >&2
+        echo ""
+        return 1
+    else
+        echo "${mnt}"
+        return 0
     fi
-    echo ""
-    return 1
 }
 
 unmount_and_release() {
-    local mnt="$1"
-    local dev="$2"
+    local dev="$1"
     local remain=3
-    if [ -n "$mnt" ]; then
-	udisksctl unmount --force -b "$dev"
-    fi
+    while true; do
+        local mnt=$(cat /proc/mounts | grep "^$dev" | cut -d' ' -f2)
+        local mnt=$(echo ${mnt} | cut -d' ' -f1)
+        if [ -n "${mnt}" ]; then
+            umount "${mnt}" > /dev/null
+            if [ "$?" -eq 1 ]; then
```

Comment:
I prefer to use `if [ ! "$?" -eq 0 ]` instead 
Please refer to the documentation: https://man7.org/linux/man-pages/man2/umount.2.html

Reply:
In keeping with usage in the rest of the script, using `if ! umount "${mnt}" > /dev/null 2>&1; then` would be even better.